backend/vkt/src/main/java/fi/oph/vkt/api/PublicController.java

```diff
@@ -43,4 +44,10 @@ public PublicReservationDTO identifyAndCreateReservation(@PathVariable final lon
 
     return publicReservationService.createReservation(examEventId, person);
   }
+
+  @DeleteMapping(path = "/reservation/{reservationId:\\d+}")
+  @ResponseStatus(HttpStatus.OK)
```

Comment:
Status taitaa defaultina olla 200 vaikka tän rivin poistaisikin

Reply:
Näinhän se näyttää olevan, otan pois. Huomasin että vastaus on aina 200 oli pyydetyllä id:llä olemassa vai ei, ei varmaan ole mitään väliä ja voidaan muuttaa myöhemmin jos on tarve.

Reply:
Aijaa.. koodista ymmärsin että heittäis virheen, jos varausta ei haetulla id:llä löydy ennen poistoa ja palauttais tyyliin 400 tai 404.

Reply:
Näin olisin itsekin arvannut. Tuolla ei varmaan mitään väliä ole tuon rajapinnan tapauksessa, mutta jännästi tulee ihan eri poikkeus kuin mitä javadoc antaa olettaa tulevan. Palauttaa nyt 404.

Reply:
Rajapinta kuitenkin jonkun virheen tais aiemminkin heittää, joka ControllerExceptionAdvicen alla handlataan ja palautetaan frontille tietty virhekoodi? Tai ainakin sen jälkeen kun kontrollerin tasolta poisti tuon, että palauttaa statuksena 200.

Muistaakseni mitä mm. AKR:n puolella testailin joskus sellaista, että kaksi välilehteä auki samalla kääntäjän sivulla ja poistaa auktorisoinnin toisesta ja yrittää tämän jälkeen saman auktorisoinnin poistoa toisesta, palauttaa palvelin jonkun virhekoodin, mutta oisko ollut 400 tai 500 404:n sijaan. Vois melkeinpä tuonne ControllerExceptionAdvice(je)n päähän tän korjauksen jollain aikavälillä toteuttaa, että handlais nämä tapaukset oikein, jos kaikissa oletettavasti sama exception class.

Reply:
Ei se alkuperäinen koodi heittänyt mitään. Se heittää jonkin springin dao poikkeuksen jos entiteetin hakee ja yrittää siitä jotakin aksessoida, tai jos pyytää poistamaan id:llä eikä sillä löydy, kuten nykyinen koodi.